src/solvers/dg_p4est/dg_2d.jl

```diff
@@ -310,8 +309,10 @@ end
 end
 
 
-function calc_surface_integral!(du, u, mesh::P4estMesh,
-                                equations, surface_integral::SurfaceIntegralWeakForm,
+function calc_surface_integral!(du, u,
+                                mesh::P4estMesh{2},
+                                equations,
+                                surface_integral::SurfaceIntegralWeakForm,
```

Comment:
@ranocha added this, I just added a line break.

Reply:
Yes, the strong form is different. The weak form only adds numerical fluxes (e.g. https://github.com/trixi-framework/Trixi.jl/blob/547170ac694ea414c51b0c31d0bc70c7681f71a1/src/solvers/dg_tree/dg_2d.jl#L1089) while the strong form adds a difference of the numerical and physical flux (e.g. https://github.com/trixi-framework/Trixi.jl/blob/547170ac694ea414c51b0c31d0bc70c7681f71a1/src/solvers/fdsbp_tree/fdsbp_2d.jl#L106-L110).

Reply:
No, I get that - I was rather asking at the technical level: Why add a type to dispatch on something where only one matching function exists in the first place? My thinking is, as long as there is no `calc_surface_integral!` for `mesh::P4estMesh{2}` and `surface_integral::SurfaceIntegralStrongFrom`, I'd argue we should not add a type annotation to `surface_integral`. That would imho defeat the purpose of when developing something new, you do not have to consider what's going on in other solvers/mesh types.

Of course, once there *is* support for strong form surface integrals for the `P4estMesh`, this type annotation makes sense and is required.

Reply:
Then feel free to remove this annotation everywhere, also for the `TreeMesh` etc. Of course, you shouldn't expect to get the strong form surface integral or helpful error messages when passing the `SurfaceIntegralStrongForm` to the DG method, of course. @efaulhaber is definitely not to blame for this since I was the one who added these additional dispatch annotations.

Reply:
Yes, yes, I know :-) Also, I am not sure which is better. I didn't realize it back when I reviewed your strong from surfint PR, but in general I feel like a feature in one part of the code should not affect features that exist (or don't) in another part of the code. I am not sure how strongly you feel about this, but we could also discuss it at the next meeting.

Reply:
Let's move the discussion to an issue since it's not really related to this PR. Could you open an issue, please, @sloede?

Reply:
Follow-up in #661, nothing to do here anymore.